java-lib/src/main/java/org/flinkextended/clink/feature/onehotencoder/ClinkOneHotEncoderModel.java

```diff
@@ -43,42 +44,59 @@
 import org.apache.flink.util.Preconditions;
 
 import com.sun.jna.Pointer;
+import org.apache.commons.compress.utils.Lists;
 import org.apache.commons.lang3.ArrayUtils;
+import org.clink.feature.onehotencoder.OneHotEncoderModelDataProto;
 import org.flinkextended.clink.jna.ClinkJna;
 import org.flinkextended.clink.jna.SparseVectorJna;
 import org.flinkextended.clink.util.ByteArrayDecoder;
 import org.flinkextended.clink.util.ByteArrayEncoder;
-import org.flinkextended.clink.util.JnaUtils;
-import org.flinkextended.clink.util.ParamUtils;
+import org.flinkextended.clink.util.ClinkReadWriteUtils;
 
 import java.io.IOException;
 import java.util.*;
-import java.util.function.Function;
-
-import static org.apache.flink.ml.util.ParamUtils.initializeMapWithDefaultValues;
+import java.util.stream.Collectors;
 
 /**
  * Wrapper class for Flink ML OneHotEncoderModel which calls equivalent C++ operator to transform.
  */
 public class ClinkOneHotEncoderModel
         implements Model<ClinkOneHotEncoderModel>, OneHotEncoderParams<ClinkOneHotEncoderModel> {
     private final Map<Param<?>, Object> paramMap = new HashMap<>();
-    private Table modelDataTable;
+    private String modelDataPath;
+    private transient Table modelDataTable;
 
     public ClinkOneHotEncoderModel() {
-        initializeMapWithDefaultValues(this.paramMap, this);
+        ParamUtils.initializeMapWithDefaultValues(this.paramMap, this);
+    }
+
+    @Override
+    public ClinkOneHotEncoderModel setModelData(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        modelDataTable = inputs[0];
+        // Newly set model data might be inconsistent with that stored in given path, thus the path
+        // must be nullified.
+        modelDataPath = null;
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {modelDataTable};
     }
 
     @Override
     public Table[] transform(Table... inputs) {
```

Comment:
Could we also add Java doc here explaining the expected behavior of this method? For example, we can say this method can only be called after the model is loaded with the path to model data), and will throw `IllegalStateException` otherwise.

Reply:
Yes, I agree with this point. What's more, It should be generally applicable to all Clink operators that transform can only be called after the load method. For now I think it is proper to add the Javadoc only to OneHotEncoderModel's transform method, but in future we can create a general interface for all such clink models and place this document there.